core/src/main/java/org/everit/email/queue/EmailQueue.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2011 Everit Kft. (http://www.everit.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.everit.email.queue;
+
+import java.util.Objects;
+
+import org.everit.email.Email;
+import org.everit.email.queue.internal.EmailQueueBulkEmailSender;
+import org.everit.email.sender.BulkEmailSender;
+import org.everit.email.sender.EmailSender;
+import org.everit.email.store.EmailStore;
+import org.everit.persistence.querydsl.support.QuerydslSupport;
+import org.everit.transaction.propagator.TransactionPropagator;
+
+/**
+ * Email Queue Sender implementation of {@link EmailQueueBulkEmailSender}.
+ */
+public class EmailQueue implements EmailSender {
+
+  private EmailQueueBulkEmailSender emailQueueBulkEmailSender;
+
+  /**
+   * Simple constructor.
+   *
+   * @param emailSender
+   *          an {@link EmailSender} instance. Cannot be <code>null</code>!
+   * @param emailStore
+   *          an {@link EmailStore} instance. Cannot be <code>null</code>!
+   * @param querydslSupport
+   *          a {@link QuerydslSupport} instance. Cannot be <code>null</code>!
+   * @param transactionPropagator
+   *          a {@link TransactionPropagator} instance. Cannot be <code>null</code>!
+   */
+  public EmailQueue(final EmailSender emailSender, final EmailStore emailStore,
```

Comment:
Create a paremter class called EmailQueueParameter. Later we can have more configuration like "parallelPassOn" if skip locked will be supported by QueryDSL.